src/TTreeWidget.cpp

```diff
@@ -255,17 +257,113 @@ void TTreeWidget::rowsInserted(const QModelIndex& parent, int start, int end)
 
         int newParentID = parent.data(Qt::UserRole).toInt();
         if (mIsTriggerTree) {
-            mpHost->getTriggerUnit()->reParentTrigger(mChildID, mOldParentID, newParentID, parentPosition, childPosition);
+            if (mOldParentID == newParentID) {
+                return;
+            }
+            if (!mParentItem) {
+                return;
+            }
+            const int m_ParentItemID = mParentItem->data(0, Qt::UserRole).toInt();
+            if (newParentID != m_ParentItemID) {
+                mParentItem = mParentItem->parent();
+            }
+            MoveTriggerCommand* command = new MoveTriggerCommand(mpHost, this, mChildID, mOldParentID, newParentID, parentPosition, childPosition, mPrevParentPosition, mPrevChildPosition);
```

Comment:
How does the memory management work here? Who will take care of deleting the `command` in the end?

Reply:
Thanks. Who will delete the `command` if it was successfully added to the undo queue?

Reply:
Thanks!